stdlib/public/runtime/SwiftDtoa.cpp

```diff
@@ -9,6 +9,11 @@
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 //===---------------------------------------------------------------------===//
+//
+// Note: This is really a C file, but Swift's build system for Linux is
+// partially allergic to C, so it's being compiled as ".cpp" for now.  Please
+// don't infect it with C++-isms.
```

Comment:
Why?

Reply:
"Don't infect with C++-isms":  Because I expect to have some discussions soon about using this code in other projects that may not be able to easily use C++.  Keeping this pure C (at least for now) will simplify possible reuse.

Plus, my test suites are predominantly pure C right now and I have a number of simplifications and/or performance improvements to the code that I'm actively testing.